include/zephyr/kernel.h

```diff
@@ -213,6 +214,37 @@ extern void k_thread_foreach_unlocked(
  */
 #define K_CALLBACK_STATE (BIT(4))
 
+#ifdef CONFIG_ARC
+/* ARC processor Bitmask definitions for threads user options */
+
+#if defined(CONFIG_ARC_DSP_SHARING)
+/**
+ * @brief DSP registers are managed by context switch
+ *
+ * @details
+ * This option indicates that the thread uses the CPU's DSP registers.
+ * This instructs the kernel to take additional steps to save and
+ * restore the contents of these registers when scheduling the thread.
+ * No effect if @kconfig{CONFIG_ARC_DSP_SHARING} is not enabled.
+ */
+#define K_DSP_IDX 6
+#define K_DSP_REGS (BIT(K_DSP_IDX))
+#endif
```

Comment:
I'd propose to define ARC-specific bits with arc prefix, K_DSP_REGS -> K_ARC_DSP_REGS.
(same for indexes and for K_AGU_REGS)